backend/src/main/kotlin/org/loculus/backend/service/files/FilesDatabaseService.kt

```diff
@@ -24,33 +26,38 @@ class FilesDatabaseService(private val dateProvider: DateProvider) {
         }
     }
 
-    fun getGroupIds(fileIds: Set<FileId>): Map<FileId, Int> =
+    fun getGroupIds(fileIds: Set<FileId>): Map<FileId, Int> = fileIds.chunkedForDatabase { chunk ->
         FilesTable.select(FilesTable.idColumn, FilesTable.groupIdColumn)
-            .where { FilesTable.idColumn inList fileIds }
-            .associate { Pair(it[FilesTable.idColumn], it[FilesTable.groupIdColumn]) }
+            .where { FilesTable.idColumn inList chunk }
+            .map { Pair(it[FilesTable.idColumn], it[FilesTable.groupIdColumn]) }
+    }.toMap()
 
     /**
      * Return a mapping of file IDs and multipart upload IDs for the files for which multipart upload has been
      * initiated but not completed
      */
-    fun getUncompletedMultipartUploadIds(fileIds: Set<FileId>): List<Pair<FileId, MultipartUploadId>> = FilesTable
-        .select(FilesTable.idColumn, FilesTable.multipartUploadId)
-        .where {
-            FilesTable.idColumn inList fileIds and
-                (FilesTable.multipartUploadId neq null) and
-                (not(FilesTable.multipartCompleted))
+    fun getUncompletedMultipartUploadIds(fileIds: Set<FileId>): List<Pair<FileId, MultipartUploadId>> =
```

Comment:
At some point we might have to worry about the map getting too bug? maybe we will have to stream - but that can probably wait until later